File: src/data/types/planetMarketState.ts
// Phase 3 GDD §2.6 -- one planet's live price state for one item.
// `currentPrice` moves via baseline drift/recovery; `basePrice` is the
// floor/ceiling reference point and the value currentPrice drifts back
// toward over time when untraded.
export interface PlanetMarketState {
  planetId: string;
  itemId: string;
  currentPrice: number;
  basePrice: number;
}

File: unity/ProfitableCore/Schema/PlanetMarketState.cs
namespace Profitable.Core.Schema;

// Ports src/data/types/planetMarketState.ts. One planet's live price state
// for one item. CurrentPrice moves via baseline drift/recovery; BasePrice
// is the floor/ceiling reference point CurrentPrice drifts back toward
// over time when untraded.
public class PlanetMarketState
{
    public string PlanetId { get; set; } = string.Empty;
    public string ItemId { get; set; } = string.Empty;
    public double CurrentPrice { get; set; }
    public double BasePrice { get; set; }
}
